tables_io/io_layer.py

```diff
@@ -358,3 +461,187 @@ def readPqToDataframes(basepath, keys=None, **kwargs):
+        The data
+    """
+    if fmt is None:
+        _, ext = os.path.splitext(filename)
+        fmt = ext[1:]
+
+    fmtlist = ['hdf5', 'parquet', 'h5']
+    if fmt not in fmtlist:
+        raise NotImplementedError(f"File format {fmt} not implemented")
+    if fmt == 'hdf5':
+        data = readHdf5ToArrayDict(filename, groupname)
+    if fmt == 'parquet':
+        data = readPqToArrayDict(filename)
+    if fmt == 'h5':
+        data = readH5ToArrayDict(filename)
+    return data
+
+
+def getGroupInputDataSize(hg):
+    """ Return the size of a HDF5 group """
+    firstkey = list(hg.keys())[0]
+    nrows = len(hg[firstkey])
+    return nrows
+
+
+def getInputDataSizeHdf5(infile, groupname='None'):
+    """ Open an HDF5 file and return  the size of a group """
+    hg, infp = readHdf5Group(infile, groupname)
+    nrow = getGroupInputDataSize(hg)
+    infp.close()
+    return nrow
+
+
+def initializeHdf5Writeout(outfile, **kwds):
+    """ Prepares an hdf5 files for parallel output
+
+    Parameters
+    ----------
+    outfile : `str`
+        The output file name
+
+    Notes
+    -----
+    The keywords should be used to create_datasets within the hdf5 file.
+
+    Each keyword should provide a tuple of ( (shape), (dtype) )
+
+    shape : `tuple` ( `int` )
+        The shape of the data for this dataset
+    dtype : `str`
+        The data type for this dataset
+
+    For exmaple
+    `initialize_writeout('test.hdf5', scalar=((100000,), 'f4'), vect=((100000, 3), 'f4'))`
+
+    Would initialize an hdf5 file with two datasets, which shapes and data types as given
+    """
+    outdir = os.path.dirname(outfile)
+    if not os.path.exists(outdir):  #pragma: no cover
+        os.makedirs(outdir, exist_ok=True)
+    outf = h5py.File(outfile, "w")
+    for k, v in kwds.items():
+        outf.create_dataset(k, v[0], v[1])
+    return outf
+
+
+def writeoutHdf5Chunk(outf, data_dict, start, end, **kwds):
+    """ Writes a data chunk to an hdf5 file
+
+    Parameters
+    ----------
+    outf : `h5py.File`
+        The file
+
+    data_dict : `dict`
+        The data being written
+
+    start : `int`
+        Starting row number to place the data
+
+    end : `int`
+        Ending row number to place the data
+
+    Notes
+    -----
+    The kwds can be used to control the output locations
+
+    For each item in data_dict, the output location is set as
+
+    `k_out = kwds.get(key, key)`
+    """
+    for key, val in data_dict.items():
+        k_out = kwds.get(key, key)
+        outf[k_out][start:end] = val
+
+
+def finalizeHdf5Writeout(outf, **kwds):
+    """ Write any last data and closes an hdf5 file
+
+    Parameters
+    ----------
+    outf : `h5py.File`
+        The file
+
+    Notes
+    -----
+    The keywords can be used to write additional data
+    """
+    for k, v in kwds.items():
+        outf[k] = v
+    outf.close()
+
+
+def iterChunkHdf5Data(infile, chunk_size=100_000, groupname='None'):
```

Comment:
TODO later - useful to be able to specify a subset of columns.

Reply:
I'll make an issue for this one.